etna/datasets/tsdataset.py

```diff
@@ -214,6 +233,8 @@ def _prepare_df(cls, df: pd.DataFrame, freq: Optional[str]) -> pd.DataFrame:
             df.index.name = index_name
 
         else:
+            if cls._check_timestamp_type(df):
+                df = cls._cast_index_to_datetime(df, freq)
             try:
                 df.index = pd.to_datetime(df.index)
```

Comment:
Let's remove this line, and replace it with calling `_cast_index_to_datetime` outside of `try` block.